From 5f1ceaad4c22d08f4fcac9db3046afbd64358249 Mon Sep 17 00:00:00 2001
From: Shannon McPherson <shannon@lunarg.com>
Date: Tue, 23 Apr 2019 10:58:19 -0600
Subject: [PATCH] vulkan: Add headless surface support to ICD header

Added the new enum value to `VkIcdWsiPlatform` and defined
`VkIcdSurfaceHeadless`
---
 include/vulkan/vk_icd.h | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/include/vulkan/vk_icd.h b/include/vulkan/vk_icd.h
index b935fa1..a2d960a 100644
--- a/include/vulkan/vk_icd.h
+++ b/include/vulkan/vk_icd.h
@@ -88,7 +88,8 @@ typedef enum {
     VK_ICD_WSI_PLATFORM_ANDROID,
     VK_ICD_WSI_PLATFORM_MACOS,
     VK_ICD_WSI_PLATFORM_IOS,
-    VK_ICD_WSI_PLATFORM_DISPLAY
+    VK_ICD_WSI_PLATFORM_DISPLAY,
+    VK_ICD_WSI_PLATFORM_HEADLESS
 } VkIcdWsiPlatform;
 
 typedef struct {
@@ -167,4 +168,8 @@ typedef struct {
     VkExtent2D imageExtent;
 } VkIcdSurfaceDisplay;
 
+typedef struct {
+    VkIcdSurfaceBase base;
+} VkIcdSurfaceHeadless;
+
 #endif  // VKICD_H
